args4j/examples/SampleMain.java

```diff
@@ -65,7 +68,7 @@ public void doMain(String[] args) throws IOException {
             // after parsing arguments, you should check
             // if enough arguments are given.
             if( arguments.isEmpty() )
-                throw new CmdLineException(parser,"No argument is given");
+                throw new CmdLineException(parser, Messages.ILLEGAL_PATH, "No argument is given");
```

Comment:
Hi,  I guess I was the guy who deprecated the 2-parameter-version:

https://github.com/kohsuke/args4j/commit/363bb4e6879c059fae7f8ffb76c141aa5932481f

 This is why I think I could bring a little light into the question whether this is right or not.

`Messages.ILLEGAL_PATH` will be output as the error and is not what you're intending to do.

One design idea of mine was to deprecate the non-localized String-only `CmdLineParserException` constructor uses in favor of localized ones. The idea besides that was the `CmdLineException`s are only instantiated from inside the args4j library.

My current usage of custom-errors is to not use exceptions for control flow but local error output code.

My suggestion would be to add Messages for typical error scenarios (which I can't name all at the moment) or do this in local error code without using the `CmdLineException`.

Hope this helps,
Stephan